src/formatDate.js

```diff
@@ -7,8 +7,58 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeperator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(fromSeperator);
+  const indexes = {
+    YYYY: fromFormat.indexOf('YYYY'),
+    YY: fromFormat.indexOf('YY'),
+    MM: fromFormat.indexOf('MM'),
+    DD: fromFormat.indexOf('DD'),
+  };
+
+  const newDate = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const part = toFormat[i];
+
+    if (part !== toFormat[toFormat.length - 1]) {
+      if (part === 'YYYY') {
+        if (dateParts[indexes['YYYY']] !== undefined) {
+          newDate.push(dateParts[indexes['YYYY']]);
+        } else {
+          newDate.push(convertYear(dateParts[indexes['YY']], 'YY'));
```

Comment:
The 'convertYear' function is called without being defined in the current scope. This is a bug as it will cause a runtime error. You need to define this function to handle the conversion of years from 'YY' to 'YYYY'.